### PR TITLE
Runs state indices

### PR DESCRIPTION
**Pull request by phixMe**
One-line summary: Adds 4 indices to help run retention faster. 

### Checklist

- [x] You've [signed-off](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#sign-your-work) your work
- [ ] Your changes are accompanied by tests (_if relevant_)
- [ ] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [ ] You've updated any relevant documentation (_if relevant_)
- [x] You've included a one-line summary of your change for the [`CHANGELOG.md`](https://github.com/MarquezProject/marquez/blob/main/CHANGELOG.md#unreleased) (_Depending on the change, this may not be necessary_).
- [ ] You've versioned your `.sql` database schema migration according to [Flyway's naming convention](https://flywaydb.org/documentation/concepts/migrations#naming) (_if relevant_)
- [ ] You've included a [header](https://github.com/MarquezProject/marquez/blob/main/CONTRIBUTING.md#copyright--license) in any source code files (_if relevant_)

### CODE REVIEW

File: api/src/main/resources/marquez/db/migration/V62__run_state_indices.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+create index run_states_uuid_index on run_states(run_uuid);
+create index parent_run_uuid_index on runs(parent_run_uuid);
+
+create index start_run_state_uuid_index on runs(start_run_state_uuid);
+create index end_run_state_uuid_index on runs(end_run_state_uuid);
```
Comment (phixMe) [inline api/src/main/resources/marquez/db/migration/V62__run_state_indices.sql:5]: Update to `if not exists`

File: api/src/main/resources/marquez/db/migration/V62__run_state_indices.sql [hunk 2]
```diff
@@ -0,0 +1,4 @@
+create index if not exists run_states_uuid_index on run_states(run_uuid);
+create index if not exists parent_run_uuid_index on runs(parent_run_uuid);
+create index if not exists start_run_state_uuid_index on runs(start_run_state_uuid);
+create index if not exists end_run_state_uuid_index on runs(end_run_state_uuid);
```
Comment (wslulciuc) [inline api/src/main/resources/marquez/db/migration/V62__run_state_indices.sql:4]: Minor: I would name the migration `V62__index_run_states_and_parent_run.sql`

Comment (phixMe) [reply api/src/main/resources/marquez/db/migration/V62__run_state_indices.sql:4]: Looks good, I'll make the change.

Comment (wslulciuc) [reply api/src/main/resources/marquez/db/migration/V62__run_state_indices.sql]: Minor: I would also make sure to prefix the runs table indices with the table name `runs_*`

**Review by wslulciuc — APPROVED** [review]
Some minor comments on naming conventions, otherwise 👍

**Pull request merged**